### PR TITLE
[mono][llvm] Remove support for llvm versions before 14.x.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -10733,11 +10733,9 @@ emit_llvm_file (MonoAotCompile *acfg)
 			opts = g_strdup_printf ("%sdefault<O2>,", opts);
 		}
 		opts = g_strdup_printf ("%splace-safepoints' -spp-all-backedges", opts);
-#elif LLVM_API_VERSION >= 1300
+#else
 		/* The safepoints pass requires the old pass manager */
```
Comment (lambdageek) [inline src/mono/mono/mini/aot-compiler.c:10737]: Didn't we upstream a port of the safe point pass to the new pass manager? I think @jandupej did it, right? Did that make it into 16.0?

**Review by lambdageek — COMMENTED** [review]
Maybe we can use the new pass manager in newer versions?

File: src/mono/mono/mini/aot-compiler.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/aot-compiler.c:10737]: This is the pre 16.x case, the 16.x case is handled above.

Comment (jandupej) [reply src/mono/mono/mini/aot-compiler.c:10737]: @lambdageek Yes, that pass should be available in 16.

**Comment by vargaz** [comment]
Might want to wait until net8 branches with this.

**Review by fanyang-mono — APPROVED** [review]
LGTM!

**Comment by SamMonoRT** [comment]
@directhex - can you confirm again we don't use 14.x or before in any of our pipelines CI and Perf. Want to avoid any lapse in coverage.

**Comment by vargaz** [comment]
Failures are unrelated

**Comment by directhex** [comment]
Not in net8 or above, unless something horrible fell through the cracks Sent from my iPhoneOn Sep 26, 2023, at 6:50 PM, Sam Patel ***@***.***> wrote:﻿
@directhex - can you confirm again we don't use 14.x or before in any of our pipelines CI and Perf. Want to avoid any lapse in coverage.

**Pull request merged**